src/main/java/com/nicusa/controller/DrugController.java

```diff
@@ -73,6 +75,70 @@
     return new ResponseEntity<>(drugAssembler.toResource(drug), HttpStatus.OK);
   }
 
+  @Transactional
+  @ResponseBody
+  @RequestMapping(value = "/portfolio/drug", method = RequestMethod.POST, consumes = "application/json")
+  public ResponseEntity<?> saveDrug(@AuthenticationPrincipal Principal principal,
+                                    @RequestBody DrugResource drugResource) {
+    Long loggedInUserProfileId = securityController.getAuthenticatedUserProfileId();
+    if (principal != null && ((UsernamePasswordAuthenticationToken)principal).getPrincipal() == loggedInUserProfileId) {
+      UserProfile userProfile = entityManager.find(UserProfile.class, loggedInUserProfileId);
+      if (userProfile == null) {
+        return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+      } else {
+        Drug drug = drugResourceToDomainConverter.convert(drugResource);
+        entityManager.persist(drug);
+        Portfolio portfolio = userProfile.getPortfolio();
+        portfolio.getDrugs().add(drug);
+        entityManager.merge(portfolio);
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.setLocation(linkTo(methodOn(DrugController.class).getPortfolioDrug(principal,
+          drug.getId())).toUri());
+        return new ResponseEntity<>(httpHeaders, HttpStatus.CREATED);
+      }
+    } else {
+      return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+    }
+  }
+
+  @Transactional
+  @ResponseBody
+  @RequestMapping(value = "/portfolio/drug/{id}", method = RequestMethod.DELETE, consumes = "application/json")
+  public ResponseEntity<?> deleteDrug(@AuthenticationPrincipal Principal principal,
+                                      @PathVariable("id") Long id) {
```

Comment:
This method should be for deleting the drug resource not managing the portfolio resource.